Sources/AccessibilitySnapshot/Core/Swift/Classes/AccessibilityHierarchyParser.swift

```diff
@@ -601,9 +601,19 @@ private extension NSObject {
                     )
                 )
             }
+
+            let explicitOrderingExceptionList = [
+                "UILayoutContainerView",
+                "UpdateCoalescingCollectionView",
+            ]
```

Comment:
Update with what I've found so far:

Both the `NavigationView` and `NavigationStack` have a `UILayoutContainerView` in their view hierarchy -- in general both view hierarchies look very similar. The key difference between the two is that the `UILayoutContainerView` in the `NavigationView` has `shouldGroupAccessibilityChildren` set to `true` while in the `NavigationStack` it is set to `false`. 

This is key to the problem because the `shouldGroupAccessiblityChildren` property is what is used to determine whether a group of accessibility elements has `explicitlyOrdered` set to `false`: https://github.com/cashapp/AccessibilitySnapshot/blob/master/Sources/AccessibilitySnapshot/Core/Swift/Classes/AccessibilityHierarchyParser.swift#L631

To me this points to what you mentioned: a larger issue around how the library is matching VoiceOver's treatment of accessibility containers. In specific I think we need improvements to how `explicitlyOrdered` is determined, or maybe how it is used to create the sorted nodes, but I don't have suggestions at this time.